src/handler.ts

```diff
@@ -88,18 +94,25 @@ router.addHandler<UserData>(
                     request.userData.site
                 )
             );
-        await Dataset.pushData(
-            reviews.map((review, i) => ({
-                ...review,
-                hotelId: request.userData.hotelId,
-                reviewPosition: request.userData.startIndex + i + 1,
-                customData: request.userData.customData,
-            }))
-        );
-        log.info(
-            `Extracted reviews ${request.userData.startIndex + 1}-${
-                request.userData.startIndex + reviews.length
-            } for hotel ${request.userData.hotelId}`
-        );
+        }
+
+        if (scrapeSettings.totalPushedResults >= scrapeSettings.maxResults) {
+            await abortIfReachedMaxResults({ request, crawler });
+        } else {
+            scrapeSettings.totalPushedResults += reviews.length;
```

Comment:
shouldn't this update be before the if(), so that we abort if the reviews from this request caused us to reach the limit?